Source/Utils/ParticleUtils.H

```diff
@@ -31,6 +32,96 @@ namespace ParticleUtils {
     findParticlesInEachCell( int const lev, amrex::MFIter const& mfi,
                              WarpXParticleContainer::ParticleTileType const& ptile);
 
+    /**
+     * \brief Return (relativistic) particle energy given velocity and mass.
+     * Note the use of `double` since this calculation is prone to error with
+     * single precision.
+     *
+     * @param[in] v2 square of particle speed ( v dot v)
+     * @param[in] mass particle mass
+     * @param[out] energy particle energy in eV
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE
+    void getEnergy ( amrex::ParticleReal const v2, double const mass,
+                     double& energy )
+    {
+        using std::sqrt;
+        using namespace amrex::literals;
+
+        // Compute c^2 since it is used repeatedly
+        constexpr auto c2 = PhysConst::c * PhysConst::c;
+        energy = mass * c2 * (sqrt(1.0_prt + v2 / c2) - 1.0_prt) / PhysConst::q_e;
```

Comment:
This form of the expression is not good when the velocity is non relativistic (since it is subject t round off). The expression can be recast as `energy = mass * u2 / (gamma + 1.) / PhysConst::q_e;`
This might fix the need for double precision mass.

Reply:
That is very nice. Thanks for pointing that out! I'll change that in a bit. 
There would still be an issue with single precision in `ParticleUtils::getCollisionEnergy` though (which is where I first observed the problem).